File: csv-machine-problems/JAVASCRIPT/mp12_tableOutput.js
// MP12 - Display Dataset in Formatted Table Output
// Student: Gray Allen Arizala

const fs = require('fs');
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question("Enter CSV dataset file path: ", function(filePath) {

    try {

        const data = fs.readFileSync(filePath, 'utf8');
        const rows = data.split('\n');

        console.log("\nFormatted Table Output:\n");

        rows.forEach(row => {

            const columns = row.split(",");

            let formattedRow = "";

            columns.forEach(col => {
                formattedRow += col.padEnd(20, " ");
            });

            console.log(formattedRow);

        });

    } catch (error) {
        console.log("Error reading dataset.");
    }

    rl.close();

});

File: csv-machine-problems/MP12_TableOutput.java
// MP12 - Display Dataset in Formatted Table Output
// Student: Gray Allen Arizala

import java.io.*;
import java.util.*;

public class MP12_TableOutput {

    public static void main(String[] args) {

        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter CSV dataset file path: ");
        String filePath = scanner.nextLine();

        try {

            BufferedReader br = new BufferedReader(new FileReader(filePath));
            String line;

            System.out.println("\nFormatted Table Output:\n");

            while ((line = br.readLine()) != null) {

                String[] columns = line.split(",");

                for (String column : columns) {

                    // Print columns with fixed spacing
                    System.out.printf("%-20s", column);
                }

                System.out.println();
            }

            br.close();

        } catch (IOException e) {
            System.out.println("Error reading dataset.");
        }
    }
}